Source/FieldSolver/MagnetostaticSolver/MagnetostaticSolver.cpp

```diff
@@ -0,0 +1,462 @@
+                }
+                else {
+                    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(false,
+                        "Field boundary conditions have to be either periodic, PEC, or neumann "
+                        "when using the magnetostatic solver"
+                    );
+                }
+
+                if ( WarpX::field_boundary_hi[idim] == FieldBoundaryType::PEC ) {
+                    if (ndotA) {
+                        hibc[adim][idim] = LinOpBCType::Neumann;
+                        dirichlet_flag[adim][idim*2+1] = false;
+                    } else {
+                        hibc[adim][idim] = LinOpBCType::Dirichlet;
+                        dirichlet_flag[adim][idim*2+1] = true;
+                    }
+                }
+                else if ( WarpX::field_boundary_hi[idim] == FieldBoundaryType::Neumann ) {
+                    hibc[adim][idim] = LinOpBCType::Neumann;
+                    dirichlet_flag[adim][idim*2+1] = false;
+                }
+                else {
+                    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(false,
+                        "Field boundary conditions have to be either periodic, PEC, or neumann "
+                        "when using the electrostatic solver"
+                    );
+                }
+            }
+        }
+    }
+    bcs_set = true;
+}
+
+void MagnetostaticSolver::EBCalcBfromVectorPotentialPerLevel::doInterp(const std::unique_ptr<amrex::MultiFab> &src,
+                                                                       const std::unique_ptr<amrex::MultiFab> &dst)
+{
+    WarpX &warpx = WarpX::GetInstance();
+
+    // Grab Interpolation Coefficients
+    // Order of finite-order centering of fields
+    const int fg_nox = WarpX::field_centering_nox;
+    const int fg_noy = WarpX::field_centering_noy;
+    const int fg_noz = WarpX::field_centering_noz;
+
+    // Device vectors of stencil coefficients used for finite-order centering of fields
+    amrex::Real const * stencil_coeffs_x = warpx.device_field_centering_stencil_coeffs_x.data();
+    amrex::Real const * stencil_coeffs_y = warpx.device_field_centering_stencil_coeffs_y.data();
+    amrex::Real const * stencil_coeffs_z = warpx.device_field_centering_stencil_coeffs_z.data();
+
+    // Synchronize the ghost cells, do halo exchange
+    ablastr::utils::communication::FillBoundary(*src,
+                                                src->nGrowVect(),
+                                                WarpX::do_single_precision_comms);
+
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (Gpu::notInLaunchRegion())
+#endif
+    for (MFIter mfi(*dst, TilingIfNotGPU()); mfi.isValid(); ++mfi)
+    {
+        IntVect const src_stag = src->ixType().toIntVect();
+        IntVect const dst_stag = dst->ixType().toIntVect();
+
+        Array4<amrex::Real const> const& src_arr = src->const_array(mfi);
+        Array4<amrex::Real> const& dst_arr = dst->array(mfi);
+
+        Box bx = mfi.growntilebox(dst->nGrowVect());
```

Comment:
I would calculate this only over the valid cells:
```suggestion
        Box bx = mfi.tilebox();
```